src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -2976,5 +2976,24 @@ public static void Main()
                 // error CS8104: An error occurred while writing the Portable Executable file.
                 Diagnostic(ErrorCode.ERR_PeWritingFailure).WithArguments("I/O error occurred.").WithLocation(1, 1));
         }
+
+        [Fact]
+        [WorkItem(9308, "https://github.com/dotnet/roslyn/issues/9308")]
+        public void FailingEmitterAllowsCancelationExceptionsThrough()
+        {
+            string source = @"
+public class X
+{
+    public static void Main()
+    {
+  
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(source);
+            var broken = new BrokenStream();
+            broken.BreakHow = 3;
```

Comment:
@TyOverby I think we should switch to using an enum values with descriptive names.